jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageReader.java

```diff
@@ -223,6 +223,26 @@
     }
   }
 
+  /**
+   * Retrieves the {@link ContainerConfigurationTemplate} for the image with the given image ID.
+   *
+   * @param imageId the image ID
+   * @return the {@link ContainerConfigurationTemplate} referenced by the image ID, if found
+   * @throws IOException if an I/O exception occurs
+   */
+  Optional<ContainerConfigurationTemplate> retrieveLocalConfig(DescriptorDigest imageId)
+      throws IOException {
+    Path configPath =
+        cacheStorageFiles.getLocalDirectory().resolve("config").resolve(imageId.getHash());
```

Comment:
Now I remember what was the risk of moving a file instead of moving a directory. Atomic move can fail (e.g., on NFS) and fall back to copying. Then it is possible that a reader opens a file that is being filled in by a writer concurrently. Does that make sense?

Reply:
Ah, yeah that makes sense.

Reply:
Hmm... on second thought this isn't any different from copying a directory recursively. A file can still be being copied. So I guess moving a file is not worse.

Reply:
Oh I just made the changes. I guess I'll revert since the former is simpler. Do you think we should go with a lockfile then, or is it rare enough to not bother?

Reply:
Yeah, I guess we are good. Not using a lockfile is currently a known flaw. :( Our assumption requires that the atmoic move should work. When it doesn't, we can't guarantee correctness.